src/textures/mlx_xpm42.c

```diff
@@ -96,8 +94,8 @@ static bool mlx_read_data(xpm_t* xpm, FILE* file, uint32_t* ctable, size_t s)
 			uint8_t* pixelstart = &xpm->texture.pixels[(y_xpm * xpm->texture.width + x_xpm) * BPP];
 			mlx_draw_pixel(pixelstart, ctable[mlx_fnv_hash(&line[x_line], xpm->cpp) % s]);
 		}
+		free(line);
```

Comment:
Yep as each line is its own allocated memory `mlx_getline` would have to be modified to use 2 `static` variables instead if you only want to allocate once. One variable would hold the size of the current string and the other the pointer. This would make it an unsafe function to be used with multithreading and sort of strange as if you would want to use the resulting line you'd have to copy it first.

Also you dont have to free the line on failure of `mlx_getline` it free's its own memory on failure (before returning `NULL`)

Reply:
If you have the energy and time to do so, go ahead!

Reply:
It now behaves like `getline` in this regard: d8bf221